src/main/webapp/script.js

```diff
@@ -88,3 +90,43 @@ function getJsonIfOk(response) {
   }
 }
 
+/**
+ * Handles response by checking it and converting it to text.
+ */
+function handleTextResponse(response) {
+  if (!response.ok) {
+    throw new Error("Response error while fetching data: " +
+        response.status + " (" + response.statusText + ")");
+  }
+  return response.text();
+}
+
+/**
+ * Loads the user's email and sets the source
+ * of the calendar iframe.
+ */
+function loadCalendar() {
+  fetch("/user")
+    .then(handleTextResponse)
+    .then(setCalendar)
+    .catch(handleNetworkError);
+}
+
+/**
+ * Sets the source of the calendar iframe including
+ * the email of the user.
+ */
+function setCalendar(email) {
+  const calendarIframe = document.getElementById("calendar");
+  calendarIframe.src = "https://calendar.google.com/calendar/embed?src=" +
+    email + "&mode=WEEK";
+}
+
+/**
+ * Refreshes the calendar iframe.
+ */
+function refreshCalendar() {
+  let calendarIframe = document.getElementById("calendar");
+  calendarIframe.src = calendarIframe.src;
```

Comment:
here it might make sense to link a resource explaining why this is the only cross-browser way to do it.
E.g. this: https://stackoverflow.com/a/4062084